### DISCUSSION TITLE
OSI Model | System Design

### DISCUSSION THREAD

**Discussion by giscus** [karanpratapsingh/portfolio#38] [Comments]
# OSI Model | System Design

In this tutorial, we will discuss OSI Model

https://www.karanpratapsingh.com/courses/system-design/osi-model

**Comment by Aztinct** [comment]
The OSI model can be seen as a universal language for computer networking. It's idea is based on splitting the entire process into 7 abstract ideas to allow shared terminologies for discussion and troubleshooting. It offers the language and structure to talk about the system of networking. 

Top to bottom
7 Application Layer. It directly interacts with data from user. Note that this isn't the same as the application that the user uses. That is software application. This application layer here is responsible for the protocols and data manipulation that the software relies on to present data to the user. This layer include HTTP (Hypertext Transfer Protocol) and SMTP (Simply Mail Transfer Protocol). HTTP is a protocol that allows communication between different systems. Commonly used for transferring data from web server to a browser to view web pages. HTTP is the protocol to allow web server to send data to client's browser. Here, it talks about how the transfer process is like. 

6 Presentation/Translation Layer. Data from application layer translated, encrypted/decrypted, compressed before it is sent out. 

5 Session Layer. Helps to open and close communication between two devices. The time between the open and close of a communication is one session. Ensure session stays open long enough to transfer data, then close it to not waste resources. 

4 Transport Layer. Does end to end communication between the two devices. 
Takes data from session layer and break it into chunks called segments. Then send it to network layer. 
Also helps to reassemble segments on receiving device into data the sesion layer can consume. 

3 Network Layer. Facilitates data transfer between two different networks. Breaks up segments into smaller units, called packets. Breaks info when sending it out. Assembles info when receiving it. It helps to find the best physical path for the data to reach its destination. This is known as routing. If two devices are communicating on the same network, then this layer is unnecessary. 

2 Data Link Layer. Facilitates data transfer between two devices on the same network. Breaks packets into smaller units called frames. 

1 Physical Layer. Physical equipment for data transfer like cables and switches. 
Data is converted into bits here, which is the 0s and 1s. Physical layer of both devices must agree on a signal convention so 1s can be distinguished from 0s on both devices. 

In summary, the 7 application layer interacts with information/data provided by end users. It decides on the way in which information is going to be exchange. Like the rules of engagement. What type of protocols to use for what type of information. The 6 translation/presentation layer is where translation, encryption/decryption, compression occurs. The information is reformatted to the required type. At 5 session layer, it helps with the opening and closing of the communication between two devices. Each open and close is considered as one session. When sending information, 4 Transport layer takes the formatted data, breaks it down into large chunks named segments, then sends it out. When it receives segments, it assembles them and pass it upwards to the upper layers to be translated, decrypted, or decompressed. 3 Network layer breaks up chunks into smaller pieces named packets to be sent out. When receiving packets, it assembles them into chunks. At this layer, routing happens and the packets are offered the best path to the receiver's device. 2 Data Link layer breaks packets into frames. 1 Physical layer refers to the physical cables and switches. Data is converted into bits here. 0s 1s are transferred between the cables. 

Questions:
- At 3 network layer, are the packets still located at the original device? Or have the packets left the device and in the midst of its journey towards the receiver's device? If yes, does this mean that segments is the info that exits the sender's device, while subsequent routers are the ones that breaks the segments into packets? 
- Why is there a need to break up the information twice? Why not once? Why not thrice? Why does the 2 Data Link layer break the packets down into frames? What does this do? 
- Why does the session have a layer of its own. I suppose it does more than open and close?
- If you access a website for 10mins, does it equate to 1 session or multiple sessions measured by when you navigate between different links on the website?

**Comment by rokito10** [comment]
Switches operates at l2
